lighthouse-core/test/computed/critical-request-chains-test.js

```diff
@@ -225,32 +244,34 @@ describe('CriticalRequestChain gatherer: extractChain function', () => {
     });
   });
 
-  it('handles redirects', () => {
-    const networkRecords = mockTracingData([HIGH, HIGH, HIGH, HIGH], [[0, 1], [1, 2], [1, 3]]);
-    const mainResource = networkRecords[0];
-
-    // Make a fake redirect
-    networkRecords[1].requestId = '1:redirected.0';
-    networkRecords[2].requestId = '1';
-
-    networkRecords[3].requestId = '2';
-    networkRecords[3].url = 'https://example.com/redirect-stylesheet';
-    networkRecords[3].resourceType = undefined;
-    networkRecords[3].statusCode = 302;
-    networkRecords[3].redirectDestination = {
-      redirectDestination: {
-        initiatorRequest: networkRecords[2],
-        resourceType: NetworkRequest.TYPES.Stylesheet,
-        priority: 'High',
-      },
-    };
+  // TODO need help
```

Comment:
any part specifically?

Reply:
revisiting this after my realization about the BFS traversal, this may be part of the problem here. a redirect request initiated by the document parser should have an edge to the redirect request and an edge to the root document but the traversalPath will only reveal the root document as the parent.